posts/how-to-efficiently-use-jhipster-in-your-company.md

```diff
@@ -0,0 +1,273 @@
+---
+authors:
+- Theo Lebrun
+categories:
+- JHipster
+date: 2018-03-23T16:12:10.000Z
+title: "How to efficiently use JHipster in your company"
+image: https://raw.githubusercontent.com/ippontech/blog-usa/master/images/2018/03/jhipster-module-header.png
+---
+
+[JHipster](http://www.jhipster.tech/) is a fantastic development platform if you want to generate, develop and deploy Spring Boot + Angular Web applications. I personnally use it to generate from proof of concept work to multiple Spring microservices. The goal of this blog post is to show how you can easily customize JHipster so it will fit your company's requirements and infrastructure. A common use case after generating a microservice is to add common dependencies, extra configuration or change the Jenkinsfile/Dockerfile.
+
+I will explain how to avoid doing manual changes after every application generation and save time.
+
+# Two ways to customize JHipster
+Since JHipster is 100% open-source, you can make all the changes you want and adapt it to your needs for free ! There are two ways to do that and I will explain below the pros/cons.
+
+SPOILER ALERT: I will show and explain how I was able to customize JHipster using a module. The module will generate a microservice with a predefined configuration (database, cache, etc) and add dependencies/templates.
+
+## Fork the main generator
+Here the [Github](https://github.com/jhipster/generator-jhipster) link, you can of course do a fork and do your custom changes directly in your new repo. That is actually the best way if you want to add new questions and customize JHipster's subgenerators. But keep in mind of something important, because JHipster is evolving every month, you will have to keep your fork up to date. That can be a lot of work especially if your changes are directly done on the original JHipster's files. Merging source and template files will take time depending on the custom changes complexity.
```

Comment:
Not sure about the formulation of "Here the Github link". Maybe "Here **is** the GitHub link"?